erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -743,3 +740,38 @@ def get_sub_assembly_items(bom_no, bom_data):
 			bom_item["stock_qty"] += d.stock_qty / d.parent_bom_qty
 
 			get_sub_assembly_items(bom_item.get("bom_no"), bom_data)
+
+@frappe.whitelist()
+def update_per_received_and_status_in_production_plan(purchase_receipt):
```

Comment:
Please follow the standard template we have - there is an extension on the gurukul group that defines it. The docstring should clearly explain any arguments and values returned by the function, if any.